### ISSUE TITLE
[BUG] Get-PnPSiteTemplate: The property or field 'ServerRelativePath'......

### ISSUE THREAD

**Issue by gabiancofiore** [pnp/powershell#145]
**Reporting an Issue or Missing Feature**
Error when getting site template with credentials (not certificate)

**Actual behavior**
Get-PnPSiteTemplate: The property or field 'ServerRelativePath' has not been initialized. It has not been requested or the request has not been executed. It may need to be explicitly requested.

**Steps to reproduce the behavior**
Connect-PnPOnline -Url $webUrl -Credentials (Get-Secret -Name $secretVault)
Get-PnPSiteTemplate -Out {filename}.xml -PersistBrandingFiles 

**Error**
Get-PnPSiteTemplate: The property or field 'ServerRelativePath' has not been initialized. It has not been requested or the request has not been executed. It may need to be explicitly requested.

**What is the version of the Cmdlet module you are running?**
Get-Module *PnP* -ListAvailable | Select-Object Name,Version | Sort-Object Version -Descending
Name                                            Version
----                                               -------
SharePointPnPPowerShellOnline  3.28.2012.0
PnP.PowerShell                             1.1.0
PnpDevice                                    1.0.0.0

$PSVersionTable
Name                           Value
----                           -----
PSVersion                      7.1.1
PSEdition                      Core
GitCommitId                    7.1.1
OS                             Microsoft Windows 10.0.21292
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

**Which operating system/environment are you running PnP PowerShell on?**
[ X ] Windows with Powershell v7.1.1

**Comment by erwinvanhunen** [comment]
Unfortunately we're not able to reproduce this issue, so it might be something in your site that is triggering this exception that we do not have in our test sites.

Please run `Get-PnPException` immediately after the error is being thrown which will show more details. 

Alternatively, can you perform the cmdlet with logging on?

```powershell
Set-PnPTraceLog -On -LogFile "C:\path\logfile.txt"
Get-PnPSiteTemplate -Out template.xml -PersistBrandingFiles
Set-PnPTraceLog -Off
```

**Comment by gabiancofiore** [comment]
Hi,
thanks for your answer, I did what you suggest and this is the result of the command and attached is the log file.

[logfile.txt](https://github.com/pnp/powershell/files/5857462/logfile.txt)


`Get-PnPException

Message          : The property or field 'ServerRelativePath' has not been initialized. It has not been requested or the request has not been executed. It may need to be explicitly requested.
Stacktrace       :    at Microsoft.SharePoint.Client.ClientObject.CheckUninitializedProperty(String propName)
                      at Microsoft.SharePoint.Client.File.get_ServerRelativePath()
                      at PnP.Framework.Provisioning.ObjectHandlers.ObjectWebSettings.PersistFile(Web web, ProvisioningTemplateCreationInformation creationInfo, PnPMonitoredScope scope, String
                   serverRelativeUrl) in C:\repos\pnpframework\src\lib\PnP.Framework\Provisioning\ObjectHandlers\ObjectWebSettings.cs:line 228
                      at PnP.Framework.Provisioning.ObjectHandlers.ObjectWebSettings.ExtractObjects(Web web, ProvisioningTemplate template, ProvisioningTemplateCreationInformation creationInfo) in       
                   C:\repos\pnpframework\src\lib\PnP.Framework\Provisioning\ObjectHandlers\ObjectWebSettings.cs:line 134
                      at PnP.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.GetRemoteTemplate(Web web, ProvisioningTemplateCreationInformation creationInfo) in
                   C:\repos\pnpframework\src\lib\PnP.Framework\Provisioning\ObjectHandlers\SiteToTemplateConversion.cs:line 137
                      at Microsoft.SharePoint.Client.WebExtensions.GetProvisioningTemplate(Web web, ProvisioningTemplateCreationInformation creationInfo) in 
                   C:\repos\pnpframework\src\lib\PnP.Framework\Extensions\WebExtensions.cs:line 1207
                      at PnP.PowerShell.Commands.Provisioning.Site.GetSiteTemplate.ExtractTemplate(XMLPnPSchemaVersion schema, String path, String packageName, ExtractConfiguration configuration) in     
                   C:\repos\powershell\src\Commands\Provisioning\Site\GetSiteTemplate.cs:line 418
                      at PnP.PowerShell.Commands.Provisioning.Site.GetSiteTemplate.ExecuteCmdlet() in C:\repos\powershell\src\Commands\Provisioning\Site\GetSiteTemplate.cs:line 149
                      at PnP.PowerShell.Commands.PnPSharePointCmdlet.ProcessRecord() in C:\repos\powershell\src\Commands\Base\PnPSharePointCmdlet.cs:line 59
ScriptLineNumber : 37
`

**Comment by erwinvanhunen** [comment]
Hi! we fixed this in the backend code of the provisioning engine. The fix will be available in release 1.2.5-nightly tomorrow.

**Comment by sainarenm564** [comment]
Hi,

I'm facing the below error while exporting template i.e., I've used Get-PnPSiteTemplate -Out template.xml

Please let me know the solution. Thanks in advance
Get-PnPSiteTemplate : Object reference not set to an instance of an object.
At C:\Users\sainaren.minnamaredd\Desktop\WPP_SP\PowerShellScripts\ExportingTemplate.ps1:10 char:1
+ Get-PnPSiteTemplate -Out template.xml
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : WriteError: (:) [Get-PnPSiteTemplate], NullReferenceException
    + FullyQualifiedErrorId : EXCEPTION,PnP.PowerShell.Commands.Provisioning.Site.GetSiteTemplate


Message          : Object reference not set to an instance of an object.
Stacktrace       :    at PnP.Framework.AuthenticationManager.<GetAccessTokenAsync>d__47.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at PnP.Framework.AuthenticationManager.<GetAccessTokenAsync>d__45.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at PnP.Framework.AuthenticationManager.<GetAccessTokenAsync>d__43.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at PnP.PowerShell.Commands.Provisioning.TokenRetrieval.<GetAccessTokenAsync>d__0.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at PnP.PowerShell.Commands.Provisioning.Site.GetSiteTemplate.<>c.<<ExtractTemplate>b__32_3>d.MoveNext()
                   --- End of stack trace from previous location where exception was thrown ---
                      at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                      at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                      at PnP.Framework.Provisioning.ObjectHandlers.PnPProvisioningContext.AcquireToken(String resource, String scope)
                      at Microsoft.SharePoint.Client.ClientContextExtensions.GetAccessToken(ClientRuntimeContext clientContext)
                      at Microsoft.SharePoint.Client.ClientContextExtensions.IsAppOnly(ClientRuntimeContext clientContext)
                      at PnP.Framework.Provisioning.ObjectHandlers.ObjectListInstance.ExtractObjects(Web web, ProvisioningTemplate template, 
                   ProvisioningTemplateCreationInformation creationInfo)
                      at PnP.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.GetRemoteTemplate(Web web, ProvisioningTemplateCreationInformation 
                   creationInfo)
                      at PnP.PowerShell.Commands.Provisioning.Site.GetSiteTemplate.ExtractTemplate(XMLPnPSchemaVersion schema, String path, String packageName, 
                   ExtractConfiguration configuration)
                      at PnP.PowerShell.Commands.Provisioning.Site.GetSiteTemplate.ExecuteCmdlet()
                      at PnP.PowerShell.Commands.PnPSharePointCmdlet.ProcessRecord()
